src/components/KBadge/KBadge.vue

```diff
@@ -1,101 +1,84 @@
     >
-      <template #content>
-        {{ truncationTooltip }}
+      <template
+        v-if="showTooltip"
+        #content
+      >
+        {{ tooltipText }}
       </template>
+      <slot
+        v-if="$slots.icon"
+        name="icon"
+      />
       <div
-        ref="badgeText"
-        class="k-badge-text"
+        ref="badgeTextElement"
+        class="badge-content-wrapper"
       >
         <slot />
       </div>
     </component>
-
-    <CloseIcon
-      v-if="dismissable"
-      :aria-hidden="hidden ? true : undefined"
-      class="k-badge-dismiss-button"
-      :color="color"
-      data-testid="k-badge-dismiss-button"
-      role="button"
-      :tabindex="hidden ? -1 : undefined"
-      @click="handleDismiss"
-      @click.stop
-    />
   </div>
 </template>
 
 <script lang="ts">
 import type { PropType } from 'vue'
-import { ref, computed, watch, useAttrs } from 'vue'
+import { ref, computed, onMounted, onBeforeUnmount } from 'vue'
 import KButton from '@/components/KButton/KButton.vue'
-import KIcon from '@/components/KIcon/KIcon.vue'
 import KTooltip from '@/components/KTooltip/KTooltip.vue'
-import type { BadgeAppearance, BadgeShape } from '@/types'
-import { BadgeAppearances, BadgeShapes } from '@/types'
+import type { BadgeAppearance } from '@/types'
+import { BadgeAppearances, BadgeMethodAppearances } from '@/types'
 import useUtilities from '@/composables/useUtilities'
-import { CloseIcon } from '@kong/icons'
+import { resizeObserverHelper } from '@/utilities/resizeObserverHelper'
 
 const { getSizeFromString } = useUtilities()
 
 // Must explicitly define components so KTooltip works in tests
 export default {
   name: 'KBadge',
-  components: { KButton, KIcon, KTooltip },
+  components: { KButton, KTooltip },
 }
 </script>
 
 <script setup lang="ts">
 const props = defineProps({
-  /**
-    * Base styling<br>
-    * One of [danger, warning, success etc.]
-    */
   appearance: {
     type: String as PropType<BadgeAppearance>,
     required: false,
+    default: 'info',
     validator: (value: string): boolean => {
-      return Object.keys({ ...BadgeAppearances }).includes(value)
+      return Object.keys(BadgeAppearances).includes(value)
     },
-    default: 'default',
   },
   /**
-   * For use with truncation. This text will be displayed
-   * on hover of the badge if the text is truncated.
+   * Tooltip text that will be displayed on hover.
    */
-  truncationTooltip: {
+  tooltip: {
     type: String,
     default: '',
   },
   /**
-   * Use this prop if you always want to show the tooltip whether
-   * or not the badge text is truncated.
-   */
-  forceTooltip: {
-    type: Boolean,
-    default: false,
-  },
-  /**
-   * Adds a dismiss button to the badge
+   * Whether tooltip should only be shown when the badge is truncated.
    */
-  dismissable: {
-    type: Boolean,
+  truncationTooltip: {
+    type: [Boolean, String], // TODO: in the next major version, remove string support
```

Comment:
question: can we just remove now?

Reply:
It doesn't hurt to have it there for backwards compatibility since the prop stays anyways.